packages/grid/_modules_/grid/themeAugmentation/overrides.ts

```diff
@@ -0,0 +1,9 @@
+import { GridClassKey } from '../models/gridClasses';
+
+export interface GridComponentNameToClassKey {
+  MuiDataGrid: GridClassKey;
+}
+
+declare module '@material-ui/core/styles/overrides' {
```

Comment:
Would it work if we stay at the public API level?
```suggestion
declare module '@material-ui/core/styles' {
```

Reply:
For `overrides` it does not work. But for `props` it works. Only `'@material-ui/core/styles'` is fine for theme props.

Reply:
We can't stay at the public API level because the `overrides` definition is not exported in https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/styles/index.d.ts. In v5 it's exported.